Tools/clinic/libclinic/parse_args.py

```diff
@@ -447,6 +470,55 @@ def parse_option_groups(self) -> None:
         parser_code = '    {option_group_parsing}'
         self.parser_body(parser_code)
 
+    def _parse_vararg(self) -> str:
+        paramname = self.varpos.converter.parser_name
+        if self.fastcall:
+            if self.limited_capi:
+                if min(self.pos_only, self.min_pos) < self.max_pos:
+                    size = f'Py_MAX(nargs - {self.max_pos}, 0)'
+                else:
+                    size = f'nargs - {self.max_pos}' if self.max_pos else 'nargs'
+                return f"""
+                    {paramname} = PyTuple_New({size});
+                    if (!{paramname}) {{{{
+                        goto exit;
+                    }}}}
+                    for (Py_ssize_t i = {self.max_pos}; i < nargs; ++i) {{{{
+                        PyTuple_SET_ITEM({paramname}, i - {self.max_pos}, Py_NewRef(args[i]));
+                    }}}}
+                    """
+            else:
+                self.codegen.add_include('pycore_tuple.h',
+                                         '_PyTuple_FromArray()')
+                if min(self.pos_only, self.min_pos) < self.max_pos:
+                    return f"""
+                        {paramname} = nargs > {self.max_pos}
+                            ? _PyTuple_FromArray(args + {self.max_pos}, nargs - {self.max_pos})
+                            : PyTuple_New(0);
+                        if ({paramname} == NULL) {{{{
+                            goto exit;
+                        }}}}
+                        """
+                else:
+                    start = f'args + {self.max_pos}' if self.max_pos else 'args'
+                    size = f'nargs - {self.max_pos}' if self.max_pos else 'nargs'
+                    return f"""
+                        {paramname} = _PyTuple_FromArray({start}, {size});
+                        if ({paramname} == NULL) {{{{
+                            goto exit;
+                        }}}}
+                        """
+        else:
+            if self.max_pos:
+                return f"""
+                    {paramname} = PyTuple_GetSlice(args, {self.max_pos}, PY_SSIZE_T_MAX);
+                    if (!{paramname}) {{{{
```

Comment:
```suggestion
                    if ({paramname} == NULL) {{{{
```

ditto (you could also prefer using `!` in the else branch instead)

Reply:
Maybe not. Actually, it affected about 850 lines, but all of them in files already touched by this PR.